pypeclub/OpenPype
# Maya: Refactor submit deadline to use AbstractSubmitDeadline

## Brief description

Refactor submit deadline to use AbstractSubmitDeadline.

## Description

Current state is functional in that it's able to do a straight maya submission (and I was able to submit a Tile Render submission, but haven't tested whether the tile assembly came out as needed yet).

I'll leave as "Draft" for the time being as I expect to still perform some fixes on my end that I can probably test easily soon.

## Additional info

Yes, this is a massive PR and likely breaks on many things - but I wanted to start raising discussions on approach. I can try to separate into standalone smaller PRs that first cleanup the old submitter and then refactor - but there will always be a tipping point where it needs to be refactored to the new logic and we'll still need to end up testing the new functionality. Thus I feel it might be worth just cleaning the monster ones and test thoroughly the different cases.

Especially because likely many cases (Tile Rendering, V-Ray submissions, Arnold submissions) were previously untested in their latest state too and might have been broken for some time.

## Testing notes:

1. Open Maya, start submitting things.
2. Open other hosts that use the AbstractDeadlineSubmitter as base - and make sure they still work too.
